src/wasm/fingerprint.rs

```diff
@@ -0,0 +1,35 @@
+use crate::Result;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use std::{fmt::Debug, path::Path};
+
+#[derive(Derivative, Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Fingerprint(u64);
+
+impl Fingerprint {
+    pub fn new(file: impl AsRef<Path> + Debug) -> Result<Fingerprint> {
+        let path = file.as_ref();
+        let meta = std::fs::metadata(path)?;
+
+        let modified = meta.modified()?;
+        let age = modified.duration_since(std::time::UNIX_EPOCH)?;
+        Ok(Self(age.as_secs()))
```

Comment:
nit: Using the time itself instead of `duration_since` would be a more unique fingerprint. Not that I'd really expect a collision either way.

Reply:
Heck, you're right! I wanted to improve the fingerprinter in general later, but it seems sufficient right now.